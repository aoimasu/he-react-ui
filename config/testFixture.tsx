import * as React from 'react';
import createTestContext from 'react-cosmos-test/enzyme';

type Fixture = {
  component: React.ComponentType<any>;
  name?: string;
};

function testOneFixture(fixture: Fixture, variant: string = 'a') {
  const { mount, getWrapper } = createTestContext({ fixture });
  const { component } = fixture;
  const { displayName, name } = component as any;
  mount();
  test(`<${displayName || name} /> rendered correctly with ${fixture.name ||
    variant} fixture`, () => expect(getWrapper()).toMatchSnapshot());
}

// TODO: use real type guard functions to discriminate one fixture vs an array of them

export default function testAllFixtures(fixtures: any, variant?: string) {
  if (!('forEach' in fixtures)) {
    testOneFixture(fixtures as Fixture, variant);
  } else if (typeof fixtures.forEach === 'function') {
    (fixtures as Fixture[]).forEach(it => testOneFixture(it));
  }
}
