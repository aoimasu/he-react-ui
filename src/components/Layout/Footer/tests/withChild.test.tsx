import fixture from '../fixtures/withChild.fixture';
import testFixture from '../../../../../config/testFixture';

testFixture(fixture, 'withChild');
