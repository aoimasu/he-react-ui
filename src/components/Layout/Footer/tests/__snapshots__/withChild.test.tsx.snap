// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> rendered correctly with withChild fixture 1`] = `
<Footer>
  <div
    className="footer"
  >
    <div>
      <span
        style={
          Object {
            "float": "left",
          }
        }
      >
        The footer is useful for creating padding at the bottom of a view.
      </span>
      <div
        style={
          Object {
            "float": "right",
          }
        }
      >
        <Button
          buttonProps={Object {}}
          color="green"
          disabled={false}
          icon="ArrowRight"
          keyline={false}
          link={false}
          newWindow={false}
          onClick={[Function]}
          squared={false}
          submit={false}
          withRouter={false}
        >
          <div
            className="buttonContainer"
          >
            <InnerButton
              buttonProps={Object {}}
              className="button green"
              newWindow={false}
              onClick={[Function]}
              submit={false}
              withRouter={false}
            >
              <button
                className="button green"
                onClick={[Function]}
                type="button"
              >
                <div
                  className="content"
                >
                  Next page
                  <Icon
                    className="legacyIcon"
                    name="ArrowRight"
                    shadow={false}
                  />
                </div>
              </button>
            </InnerButton>
          </div>
        </Button>
      </div>
    </div>
  </div>
</Footer>
`;
